components/ruby/lib/license_acceptance/strategy/file.rb

```diff
@@ -29,11 +29,23 @@ def accepted?(product_relationship)
 
         searching.each do |product|
           config.license_locations.each do |loc|
-            f = ::File.join(loc, product.filename)
-            if ::File.exist?(f)
-              logger.debug("Found license #{product.filename} at #{f}")
-              missing_licenses.delete(product)
-              break
+            license_path = ::File.join(loc, product.filename)
+            if ::File.exist?(license_path)
+              ::File.open(license_path, ::File::RDONLY) do |license_file|
+                license = YAML.load(license_file.read)
+
+                # If the license file is missing the license_name field,
+                # assume it represents the EULA
+                license["license_name"] ||= "EULA"
+
+                if product.license.name == license["license_name"]
+                  # It is possible to have previously accepted a product under the EULA license but that
+                  # product is actually covered under MLSA, so we need to check it.
+                  logger.debug("Found license #{product.filename} at #{license_path}")
+                  missing_licenses.delete(product)
+                  break
+                end
+              end
```

Comment:
`YAML.load_file` could be used here to avoid a level of nesting and some file boilerplate.

```suggestion
              license = YAML.load_file(license_path)

              # If the license file is missing the license_name field,
              # assume it represents the EULA
              license["license_name"] ||= "EULA"

              if product.license.name == license["license_name"]
                # It is possible to have previously accepted a product under the EULA license but that
                # product is actually covered under MLSA, so we need to check it.
                logger.debug("Found license #{product.filename} at #{license_path}")
                missing_licenses.delete(product)
                break
              end
```